### PR TITLE
Update guidance on changing a course or course location

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

The support tech docs need to be updated to include the correct way to change a course and or course location in place (bypassing the state machine)

## Changes proposed in this pull request

- Update support docs 

## Guidance to review

- Make sense? 
- Are there still scenarios when we would prefer to use the old method?

## Link to Trello card

https://trello.com/c/2EucQz0o/4150-update-support-tech-docs

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by despo — DISMISSED** [review]
👍🏼

File: docs/support_playbook.md [hunk 1]
```diff
@@ -98,62 +98,56 @@ Make sure you know which part you are amending. Add `\r\n\r\n` for carriage retu
...
 ```
 
-- Find the application form
-
-Eg -
+- Find the associated course option 
 
 ```ruby
-application_form = old_application_choice.application_form
+new_course_option = CourseOption.where(course_id: new_course.id).first
```
Comment (despo) [inline docs/support_playbook.md:126]: A course has multiple course options assigned to it, so this is not quite right as we are randomly just picking the first one.

Comment (duncanjbrown) [reply docs/support_playbook.md:126]: yes — we should depend on a `CourseOption`, not a `Course`.

**Review by duncanjbrown — COMMENTED** [review]
This is better. Would it be even better as a function (ie a service)? Which we could then test.

**Comment by TobyRet** [comment]
> This is better. Would it be even better as a function (ie a service)? Which we could then test.

There is a card on the Provendor board for this @duncanjbrown - https://trello.com/c/LglxncQF/4520-create-support-tooling-for-updating-a-candidates-course-and-or-course-location

If this type of request is likely to increase then perhaps it's worth adding tooling to the UI so support handle this themselves

File: docs/support_playbook.md [hunk 2]
```diff
@@ -98,61 +98,57 @@ Make sure you know which part you are amending. Add `\r\n\r\n` for carriage retu
...
-course_option = CourseOption
+new_course_option = CourseOption
+  .includes(:site)
   .where(
-    course_id: Course.where(code: _course_code, provider: Provider.find_by(code: _provider_code)),
-    recruitment_cycle_year: RecruitmentCycle.current_year
+    course_id: course,
+    study_mode: _study_mode,
+    site: { id: _site_id }
   )
```
Comment (cwrw) [inline docs/support_playbook.md:128]:
```suggestion
  ).first
```

File: docs/support_playbook.md [hunk 3]
```diff
@@ -98,61 +98,57 @@ Make sure you know which part you are amending. Add `\r\n\r\n` for carriage retu
...
-### Make or change offer
+## Courses and course locations 
 
-If the current application status is `awaiting_provider_decision` use MakeAnOffer service.
-
-If the current application status is `offer` use ChangeOffer service.
+### Changing a course or course location 
 
-### Change provider/course
+A provider may request that a candidate be placed on an different course, or a different site.
```
Comment (cwrw) [inline docs/support_playbook.md:105]: could we indicate that this an application in `awaiting_provider_decision` and not already in the offered state?

**Review by cwrw — COMMENTED** [review]
looks good, thanks @TobyRet. Tooling is coming, this is just as until we get that up and running to avoid losing context

File: docs/support_playbook.md [hunk 4]
```diff
@@ -98,61 +98,57 @@ Make sure you know which part you are amending. Add `\r\n\r\n` for carriage retu
...
-- Create a new ApplicationChoice associated with the application form
-
-Then submit the new application choice to the provider by calling a service:
+- Finally, update the course option
 
 ```ruby
-SupportInterface::AddCourseChoiceAfterSubmission.new(application_form: application_form, course_option: course_option).call
+application_choice.update_course_option_and_associated_fields!(
+  new_course_option, 
+  audit_comment: "Zendesk ticket #_ticket_number - update course")
```
Comment (despo) [inline docs/support_playbook.md:142]: I was trying to put this into a service and I think this method was put in place in order to easily update the offered_course and not the applied one, so we need to also specify `course_option` as part of other fields in order to get the course to be updated sucesfuly.


```suggestion
application_choice.update_course_option_and_associated_fields!(
  new_course_option, 
  other_fields: { course_option: new_course_option },
  audit_comment: "Zendesk ticket #_ticket_number - update course")
```

File: docs/support_playbook.md (hunk 3 above)
Comment (cwrw) [reply docs/support_playbook.md:105]: ah also I forgot about applications in the interviewing state, do we want to add something there?

Comment (TobyRet) [reply docs/support_playbook.md]: I've made the change @cwrw 

But given we now have a service in progress (https://github.com/DFE-Digital/apply-for-teacher-training/pull/6083) this guidance will become out of date shortly. @despo shall I close this PR?

File: docs/support_playbook.md (hunk 4 above)
Comment (TobyRet) [reply docs/support_playbook.md:143]: 🤔 When I did some of these tickets I did not pass in `other_fields: { course_option: new_course_option }` - everything seemed to be ok. If 'other_fields' is a required param in this scenario then seems like this should not be 'optional'?  Perhaps this is a sign we need a different method with a different (more explicit) signature?

**Comment by despo** [comment]
Closing in favor of #6116

**Pull request closed without merging**